### PR TITLE
Config file

### PR DESCRIPTION
**Pull request by najeal**
Adding config file validation with machine name checking

#125

### CODE REVIEW

File: pkg/config/machine.go [hunk 1]
```diff
@@ -51,3 +58,17 @@ type Machine struct {
 	// Cmd is a cmd which will be run in the container.
 	Cmd string `json:"cmd,omitempty"`
 }
+
+// validate checks basic rules for Machine's fields
+func (conf Machine) validate() error {
+	errMessage := "Machine configuration not valid"
+	matching, err := regexp.MatchString(".*%d", conf.Name)
```
Comment (dlespiau) [inline pkg/config/machine.go]: `strings.Contains` is enough here.

File: pkg/config/machine.go [hunk 2]
```diff
@@ -51,3 +58,17 @@ type Machine struct {
...
+	if err != nil {
+		log.Warnf(err.Error())
+		return fmt.Errorf(errMessage)
+	} else if matching == false {
+		log.Warnf("Machine conf validation: machine name %v is not valid, it should finish with %%d", conf.Name)
```
Comment (dlespiau) [inline pkg/config/machine.go]: `%d` is actually valid anywhere in the string not just at the end.

File: tests/test-create-invalid-fedora29.cmd [hunk 3]
```diff
@@ -0,0 +1 @@
+footloose create --config %testName.static
```
Comment (dlespiau) [inline tests/test-create-invalid-fedora29.cmd]: Just a note: while it's ok to have one integration test to make sure the validation is bubbling up to the user, validation is one of those things that should be done with unit tests (not now, it's ok to add them when we have more than one check).

Comment (najeal) [reply tests/test-create-invalid-fedora29.cmd]: I take note

**Comment by dlespiau** [comment]
All green! Thanks for your PRs, good work! :)

**Pull request merged**